packages/analytics/amplify_analytics_pinpoint_dart/lib/src/impl/analytics_client/analytics_client.dart

```diff
@@ -41,13 +39,21 @@ class AnalyticsClient {
   final DeviceContextInfoProvider? _deviceContextInfoProvider;
   final EndpointInfoStoreManager _endpointInfoStoreManager;
 
-  /// Allows mocking of [PinpointClient] for unit tests
-  @visibleForTesting
-  Future<void> initWithClient({
+  /// Async initialize the [eventClient] and [endpointClient].
+  /// Requires missing values typically provided in plugin configure.
+  Future<void> init({
     required String pinpointAppId,
-    required PinpointClient pinpointClient,
+    required String region,
+    required AWSCredentialsProvider authProvider,
     QueuedItemStore? eventStore,
   }) async {
+    final pinpointClient = PinpointClient(
+      // ignore: invalid_use_of_protected_member, invalid_use_of_visible_for_testing_member
+      client: AmplifyClass.instance?.dependencies.get<AWSHttpClient>(),
```

Comment:
```suggestion
      client: Amplify.dependencies.getOrCreate<AmplifyHttpClient>(),
```

Reply:
But this should be passed to each SDK call, not to the constructor.

Reply:
Is that necessary though?  Since PinpointClient uses that instance to make each call: `client: client ?? _client,`

Reply:
Yes, because the following should work:

```dart
final analyticsClient = AnalyticsClient();
await analyticsClient.init();
Amplify.dependencies.addInstance(MockAmplifyHttpClient());
await analyticsClient.eventClient.recordEvent(...); // should use `MockAmplifyHttpClient`
```

Reply:
Ah shoot of course, thank you